### PR TITLE
Fixes for pr190

### PR DESCRIPTION
**Pull request by DanRStevens**
This is set to merge into PR #190.

This contains fixes for a number of the points I brought up, plus a few exception message fixes I noticed while testing.

I'm putting this in a new branch, in case there are contentious sections.

### CODE REVIEW

File: srcDLL/DllMain.cpp [hunk 1]
```diff
@@ -89,8 +89,8 @@ int TApp::Init()
 	// Find VOL files from additional folders
 	for (std::size_t i = 0; i < moduleLoader->Count(); ++i) {
 		std::string moduleDirectory = moduleLoader->GetModuleDirectory(i);
-		if (moduleDirectory != std::string()) {
-			LocateVolFiles();
+		if (!moduleDirectory.empty()) {
```
Comment (Brett208) [inline srcDLL/DllMain.cpp]: I like this improved presentation.

File: srcDLL/DllMain.cpp [hunk 2]
```diff
@@ -89,8 +89,8 @@ int TApp::Init()
...
+			LocateVolFiles(moduleLoader->GetModuleName(i));
```
Comment (Brett208) [inline srcDLL/DllMain.cpp]: Bad mistake on my part.

File: srcStatic/GameModule.h [hunk 3]
```diff
@@ -35,6 +35,11 @@ class GameModule
 	// Console module calls: mod_run
 	virtual void Run() { };
 
+	// Returns the directory associated with this module
+	// Can be used to access additional resources packaged with the module
+	// By default return the empty string (implying no additional resources)
+	virtual std::string Directory() { return std::string(); }
```
Comment (Brett208) [inline srcStatic/GameModule.h]: I'm not sure that I prefer this over the dynamic_cast, but not too big a deal either way...

File: srcStatic/GameModules/ConsoleModule.cpp [hunk 4]
```diff
@@ -14,19 +14,13 @@ ConsoleModule::ConsoleModule(const std::string& moduleName) : DllModule(moduleNa
 			moduleDirectory + " : " + GetLastErrorString());
 	}
 
-
-	try {
```
Comment (Brett208) [inline srcStatic/GameModules/ConsoleModule.cpp]: Remove this try/catch for module loading should also be removed from the ini module code.

Comment (DanRStevens) [reply srcStatic/GameModules/ConsoleModule.cpp]: Ahh, good catch. Go for it.

Comment (DanRStevens) [reply srcStatic/GameModules/ConsoleModule.cpp]: Just a reminder this suggestion seems to have been left out.

Comment (Brett208) [reply srcStatic/GameModules/ConsoleModule.cpp]: I looked at this closer the other day and thought it wasn't warranted. Bad suggestion on my part.

**Pull request merged**